### PR TITLE
MCP setup bug on empty JSON file

### PR DESCRIPTION
**Pull request by KevyVo**
## Description
Currently, when we run `defang mcp setup`, the command attempts to write the config. One edge case I didn’t account for is when an mcp.json file already exists but is empty or just has a bunch of white spaces. Right now, we don’t overwrite it (though we should, since it’s empty). If the file exists and contains invalid JSON, we just return an error.

I also updated the messaging to tell user to update their IDE to the new version for the best experenice.

This PR will fix the mention edge case.
Screenshot 2025-09-19 at 5 19 06 PM

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added appropriate tests
- [ ] I have updated the Defang CLI docs and/or README to reflect my changes, if necessary

### CODE REVIEW

File: src/pkg/mcp/setup.go [hunk 1]
```diff
@@ -259,9 +259,15 @@ func handleVSCodeConfig(configPath string) error {
...
 	if data, err := os.ReadFile(configPath); err == nil {
-		// File exists, parse it
-		if err := json.Unmarshal(data, &existingData); err != nil {
-			return fmt.Errorf("failed to unmarshal existing vscode config %w", err)
+		// Check if file is empty or only contains whitespace
+		if len(strings.TrimSpace(string(data))) == 0 {
+			// File is empty, treat as new config
+			existingData = make(map[string]any)
+		} else {
+			// File exists, parse it
```
Comment (KevyVo) [inline src/pkg/mcp/setup.go:267]:
```suggestion
			// File is not empty, attempt to parse it
```

File: src/pkg/mcp/setup.go [hunk 2]
```diff
@@ -259,9 +259,15 @@ func handleVSCodeConfig(configPath string) error {
 
 	// Check if the file exists
 	if data, err := os.ReadFile(configPath); err == nil {
-		// File exists, parse it
-		if err := json.Unmarshal(data, &existingData); err != nil {
-			return fmt.Errorf("failed to unmarshal existing vscode config %w", err)
+		// Check if file is empty or only contains whitespace
+		if len(strings.TrimSpace(string(data))) == 0 {
```
Comment (lionello) [inline src/pkg/mcp/setup.go:263]:
```suggestion
		if len(bytes.TrimSpace(data)) == 0 {
```

File: src/pkg/mcp/setup_test.go [hunk 3]
```diff
@@ -593,6 +611,22 @@ func TestWriteConfig(t *testing.T) {
 			existingData:  `{"mcpServers": "not an object}`,
 			expectedError: true,
 		},
+		{
+			name:         "standard_config_new_file_empty",
+			fileExists:   true,
+			existingData: "",
```
Comment (lionello) [inline src/pkg/mcp/setup_test.go:617]: add a case for `"  "`?

**Review by lionello — APPROVED** [review]
